src/main/java/uk/gov/hmcts/reform/sscs/config/HmcHearingsEventTopicListener.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.sscs.config;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusException;
+import com.azure.messaging.servicebus.ServiceBusFailureReason;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.sscs.model.HmcMessage;
+
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@Slf4j
+@Data
+@Configuration
+public class HmcHearingsEventTopicListener {
+
+    @Value("${azure.service-bus.connectionString}")
+    private String connectionString;
+    @Value("${azure.service-bus.topicName}")
+    private String topicName;
+    @Value("${azure.service-bus.subscriptionName}")
+    private String subscriptionName;
+
+    //TODO add @Bean and add correct values (connectionString,topicName,subscriptionName) in application.yml
```

Comment:
Still need some way for this to actually be instantiated. I think @Configuration at class level and @Bean at method level should do.

Reply:
must have a subscriptionName associated with it.

Reply:
It doesn't need to have one, I've seen multiple implementations without it. The connection string and queue name should encode all the information

Reply:
Ignore me on that one, the topics do require subscription names - you're right